### PR TITLE
Fix ebeanModels errors

### PR DESCRIPTION
**Pull request by jroper**
* Ensure that when there is an error loading ebeanModels, the exception is correctly reported, not lost by classloader issues.
* Allow ebeanModels to run even when there is no application.conf file.

### CODE REVIEW

File: play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
     @Override
     public  Map<String, List<String>> apply(ClassLoader classLoader) {
-        Environment env = new Environment(new File("."), classLoader, Mode.PROD);
+        // Using TEST mode is the only way to load configuration without failing if application.conf doesn't exist
+        Environment env = new Environment(new File("."), classLoader, Mode.TEST);
         Configuration config = Configuration.load(env);
```
Comment (richdougherty) [inline play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java]: My recommendation would be to use another `load` method that lets you set the `allowMissingApplicationConf` option.

```scala
Configuration.load(environment.classLoader, System.getProperties, Map.empty, allowMissingApplicationConf = allowMissingApplicationConf = true)
```

File: sbt-play-ebean/src/main/scala/play/ebean/sbt/SbtEbean.scala [hunk 2]
```diff
@@ -111,6 +111,20 @@ object SbtEbean extends AutoPlugin {
...
+          // and stack trace
+          def clone(t: Throwable): RuntimeException = {
+            val cloned = new RuntimeException(s"${t.getClass.getName}: ${t.getMessage}")
+            cloned.setStackTrace(t.getStackTrace)
+            if (t.getCause != null) {
+              cloned.initCause(clone(t.getCause))
+            }
+            cloned
+          }
+          throw clone(e)
```
Comment (richdougherty) [inline sbt-play-ebean/src/main/scala/play/ebean/sbt/SbtEbean.scala]: That should do the trick!

File: play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java (hunk 1 above)
Comment (jroper) [inline play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java]: I tried that first.  It's not accessible from Java.  I don't think the Scala compiler generates static methods for private methods.

Comment (jroper) [inline play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java]: Note, this is Java code here, not Scala.

**Pull request merged**